src/librbd/io/ImageRequest.cc

```diff
@@ -643,6 +659,11 @@ int ImageDiscardRequest<I>::prune_object_extents(
                      remove_lambda),
       object_extents->end());
   }
+
+  if (length_modified) {
```

Comment:
TBF I have no clue what I am doing here, this is all just guess-work (I started to look at the codebase properly a month ago and my C++ knowledge is a decade old). I rather heavily depend on you figuring out edge-cases and me just pointing out the obvious flaw.

I was wondering about more issues around this and it would make sense to `return 1` if anything is modified (as you mentioned), thus
```diff
+
+    if (length_modified || prune_required) {
+      return 1;
+    }
```

Reply:
I updated the code to only trigger return 1 if
* prune_required
* length was actually modified (we don't check this earlier)